### ISSUE TITLE
Loading the pretrained model

### ISSUE THREAD

**Issue by LeilaMaria** [wanghao9610/TMANet#26]
1.I have a question concerning loading the camvid pretrained model. Once it is loaded, I am seeing the weights for ResNet architecture and I am seeing the two last tensors (fc.weight and fc.bias). These two tensors loaded are related to what? Are they related to the segmentation head or to which part of the network because I can't see the relation with ResNet??

2. In the paper, it is mentioned that "we add the auxiliary segmentation loss at the low-level feature of the backbone (e.g. the stage 3 output of ResNet)", this means that it is added at the output of layer3 of ResNet? It is not clear for me? Could you plz clarify it?

Best Regards,

**Comment by wanghao9610** [comment]
Hi, @LeilaMaria . For you first qustion, fc.weight and fc.bias is the last linear layer on ResNet for image classification, in segmentation model, ResNet is a backbone to extract features and the weights of last linear layer almost not been loaded. As for you sencond question, the auxilary loss on segmentation, which utilizes a 1x1 conv layer following the ResNet stage 3 feature to output the logits, and use the CE_loss to calculate the loss between the logits and groundtruth. Hope it would be helpful.

**Comment by LeilaMaria** [comment]
@wanghao9610 Thank you for considering my question.
1. So, when we load the model, we are getting the weights of the last fully connected layers of ResNet, but we are not using them in this algo?? Because there is no fully connected layers here in ResNet, they are in the auxiliary head.
2.Okay, but where can I find the auxiliary loss in the algo? Because when I load the architecture of the model I can't see it and I am searching the models script to find where it is exactly mentioned ? I al searching backbones/ResNet and the scripts to find it.

Best Rrgards,

**Comment by wanghao9610** [comment]
Let's fix your confusion one by one. 
1. It is common usage that PyTorch loading model, where PyTorch will load the model weights depending the state_dict (it is a key and value pair, key is the layer_name, value is the weight). That is to say, althogh we load the pre-trained weight, but some weight doesn't match with our model and not be used (e.g. the classification layers of ResNet). 
2. You can reference the [config_file ](https://github.com/wanghao9610/TMANet/blob/fa6a27a75dfedd14e9ba7f82abbde2b7b47d34d1/configs/_base_/models/tmanet_r50-d8.py#L31)to find the auxiliary_head module.
Hope it can solve your confusion.

**Comment by LeilaMaria** [comment]
Thank you for considering my question. 

1.I just want to make sure that everything is clear for me. If I am working with Camvid, the output of the concatenated sequences after Mv and Qv has a shape of [B,512,80,80]. Therefore the conv_seg speficified in TMAHead is the one classifying the output conv2d(512,12,...). While the auxiliary head is the one connected directly to the stage 3 of ResNet?

2.Therefore, the segmentation head after C (concatenate symbol) in you schema (fig. 2 of the paper) is mainly formed by conv_seg(512,... in TMAHead?

**Comment by wanghao9610** [comment]
May be you can debug to print the shape of features that you want to make sure, it is the most strightforward to do.

**Comment by LeilaMaria** [comment]
okay, thank you for your answer.
